R/temporal_torus_translation.R

```diff
@@ -14,16 +14,41 @@
 #' # Calculate a covariance matrix on a null community
 #' data(knz_001d)
 #' a1_cyclic <- cyclic_shift(subset(knz_001d, subplot=="A_1"), time.var="year",
-#' species.var="species", abundance.var="abundance", FUN=cov, bootnumber = 10)
+#' species.var="species", abundance.var="abundance", method=cov, bootnumber = 10)
 #' @references
 #' Hallett, Lauren M., Joanna S. Hsu, Elsa E. Cleland, Scott L. Collins, Timothy L. Dickson, Emily C. Farrer, Laureano A. Gherardi, et al. "Biotic Mechanisms of Community Stability Shift along a Precipitation Gradient." Ecology 95, no. 6 (2014): 1693-1700.
 #'
 #' Harms, Kyle E., Richard Condit, Stephen P. Hubbell, and Robin B. Foster. "Habitat Associations of Trees and Shrubs in a 50-Ha Neotropical Forest Plot." Journal of Ecology 89, no. 6 (2001): 947-59.
 #' @export
-cyclic_shift <- function(df, time.var="year", species.var="species",  abundance.var="abundance", FUN, bootnumber){
-  if(!is.numeric(df[[abundance.var]])) { stop("Abundance variable is not numeric") }
-  out<-replicate(bootnumber, FUN(shuffle_community(transpose_community(df, time.var,  species.var, abundance.var))))
-  return(out)
+cyclic_shift <- function(df, time.var="year",
+                         species.var="species",
+                         abundance.var="abundance",
+                         method,
+                         bootnumber){
+
+  ## match arg to check method name
+  ## switch to go from character to function
```

Comment:
Hi Andrew - I think we want it to be an actual function - with the
restructure this will be a pretty general function, people might want to
use it. When its used in the variance_ratio it'll always be
variance_ratio_matrixdata

On Fri, Mar 18, 2016 at 12:59 AM, Andrew MacDonald <notifications@github.com
> wrote:

> In R/temporal_torus_translation.R
> <https://github.com/laurenmh/codyn/pull/67#discussion_r56619026>:
>
> >  #' @references
> >  #' Hallett, Lauren M., Joanna S. Hsu, Elsa E. Cleland, Scott L. Collins, Timothy L. Dickson, Emily C. Farrer, Laureano A. Gherardi, et al. "Biotic Mechanisms of Community Stability Shift along a Precipitation Gradient." Ecology 95, no. 6 (2014): 1693-1700.
> >  #'
> >  #' Harms, Kyle E., Richard Condit, Stephen P. Hubbell, and Robin B. Foster. "Habitat Associations of Trees and Shrubs in a 50-Ha Neotropical Forest Plot." Journal of Ecology 89, no. 6 (2001): 947-59.
> >  #' @export
> > -cyclic_shift <- function(df, time.var="year", species.var="species",  abundance.var="abundance", FUN, bootnumber){
> > -  if(!is.numeric(df[[abundance.var]])) { stop("Abundance variable is not numeric") }
> > -  out<-replicate(bootnumber, FUN(shuffle_community(transpose_community(df, time.var,  species.var, abundance.var))))
> > -  return(out)
> > +cyclic_shift <- function(df, time.var="year",
> > +                         species.var="species",
> > +                         abundance.var="abundance",
> > +                         method,
> > +                         bootnumber){
> > +
> > +  ## match arg to check method name
> > +  ## switch to go from character to function
>
> I was thinking of allowing people to specify their method using
> characters, rather than passing in the actual function . but perhaps
> unnecessary ? isn't method always going to be variance_ratio_longdata or
> variance_ratio_matrixdata ? do we even *want* users to use this function
> anymore?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/laurenmh/codyn/pull/67/files/f3e38639882cb3e07b489b62de4bd6214853b215#r56619026>
>